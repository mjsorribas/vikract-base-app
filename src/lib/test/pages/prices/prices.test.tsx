import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Prices from "lib/pages/prices/index";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("Prices Page", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/prices" }]}>
      <Prices />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
