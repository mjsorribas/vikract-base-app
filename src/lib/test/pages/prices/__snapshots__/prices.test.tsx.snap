// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Prices Page 1`] = `
<div
  className="css-qi60vh"
>
  <section
    className="css-137ocyi"
  >
    <div
      className="css-1o5einl"
    >
      <div
        className="css-1s8shn8"
      >
        <div
          className="chakra-stack css-ylms7g"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-2cog7p"
            fill="currentColor"
            focusable={false}
            height="1em"
            role="img"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M0 0v11.408h11.408V0zm12.594 0v11.408H24V0zM0 12.594V24h11.408V12.594zm12.594 0V24H24V12.594z"
            />
          </svg>
          <h2
            className="chakra-heading css-appnyl"
          >
            Basic UI
          </h2>
        </div>
        <div
          className="css-d34znr"
        >
          <h2
            className="chakra-heading css-vty8mk"
          >
            $29
          </h2>
          <p
            className="chakra-text css-1eruqsj"
          >
            / yr
          </p>
        </div>
        <ul
          className="css-1gf044s"
          role="list"
        >
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Basic application UI components
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Lifetime access for 6 months
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Use on your project
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Free Updates for 6 months
          </li>
        </ul>
        <a
          className="chakra-link css-0"
          href="#"
        >
          <button
            className="chakra-button css-1sjc8a"
            type="button"
          >
            Buy now
          </button>
        </a>
      </div>
      <div
        className="css-mek8o5"
      >
        <div
          className="css-str4sf"
        >
          <p
            className="chakra-text css-1fkr1na"
          >
            Popular
          </p>
        </div>
        <div
          className="chakra-stack css-ylms7g"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-2cog7p"
            fill="currentColor"
            focusable={false}
            height="1em"
            role="img"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M19.442 21.355c.55-.19.74-.256.99-.373.342-.152.605-.39.605-.818a.846.846 0 00-.605-.813c-.318-.092-.703.042-.99.122l-5.42 1.46a7.808 7.808 0 01-4.057 0l-5.407-1.46c-.287-.08-.672-.214-.99-.122a.847.847 0 00-.605.813c0 .427.263.666.605.818.25.117.44.184.99.373l5.138 1.79c1.491.52 3.104.52 4.601 0zm-9.263-3.224a7.622 7.622 0 003.636 0l8.01-1.967c.507-.122.709-.165.99-.257.354-.116.605-.415.605-.806a.847.847 0 00-.605-.813c-.281-.08-.697.024-.99.08l-8.664 1.545a6.813 6.813 0 01-2.334 0l-8.652-1.545c-.293-.056-.708-.16-.99-.08a.847.847 0 00-.604.813c0 .39.25.69.604.806.282.092.483.135.99.257zM14.75.621a24.43 24.43 0 00-5.511 0L6.495.933c-.294.03-.715.055-.99.14-.28.092-.605.355-.605.807 0 .39.257.702.605.806.281.08.696.074.99.074h11.01c.293 0 .709.006.99-.074a.835.835 0 00.605-.806c0-.452-.324-.715-.605-.807-.275-.085-.697-.11-.99-.14zm6.037 6.767c.3-.019.709-.037.99-.116a.84.84 0 000-1.614c-.281-.085-.69-.073-.99-.073H3.214c-.3 0-.709-.012-.99.073a.84.84 0 000 1.614c.281.079.69.097.99.116l7.808.556c.642.042 1.308.042 1.943 0zm1.62 4.242c.513-.08.708-.104.989-.202.354-.121.605-.409.605-.806a.84.84 0 00-.605-.806c-.28-.086-.69-.019-.99.012l-9.232.929c-.776.079-1.582.079-2.358 0l-9.22-.93c-.3-.03-.715-.097-.99-.011a.84.84 0 00-.605.806c0 .397.25.685.605.806.275.092.476.123.99.202l8.823 1.418c1.038.165 2.12.165 3.158 0Z"
            />
          </svg>
          <h2
            className="chakra-heading css-appnyl"
          >
            Professional UI
          </h2>
        </div>
        <div
          className="css-d34znr"
        >
          <h2
            className="chakra-heading css-vty8mk"
          >
            $49
          </h2>
          <p
            className="chakra-text css-1eruqsj"
          >
            / yr
          </p>
        </div>
        <ul
          className="css-1gf044s"
          role="list"
        >
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Professional application UI
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Lifetime access for 1 year
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Use on 1 project
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Free Updates for 1 year
          </li>
        </ul>
        <a
          className="chakra-link css-0"
          href="#"
        >
          <button
            className="chakra-button css-1sjc8a"
            type="button"
          >
            Buy now
          </button>
        </a>
      </div>
      <div
        className="css-1s8shn8"
      >
        <div
          className="chakra-stack css-ylms7g"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-2cog7p"
            fill="currentColor"
            focusable={false}
            height="1em"
            role="img"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M16.146 0v24l6.134-4.886V3.334zM13.293 18.758l-4.939 2.157V2.086l4.939 1.462zM1.721 18.205l3.78-.999V5.188l-3.762-.606z"
            />
          </svg>
          <h2
            className="chakra-heading css-appnyl"
          >
            Enterprise UI
          </h2>
        </div>
        <div
          className="css-d34znr"
        >
          <h2
            className="chakra-heading css-vty8mk"
          >
            $99
          </h2>
          <p
            className="chakra-text css-1eruqsj"
          >
            / yr
          </p>
        </div>
        <ul
          className="css-1gf044s"
          role="list"
        >
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Enterprise application UI
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Lifetime access for 2 year
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Use on 1 project
          </li>
          <li
            className="css-1xnzs7r"
          >
            <svg
              aria-hidden="true"
              className="chakra-icon css-1v19hek"
              fill="currentColor"
              focusable={false}
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                fillRule="evenodd"
              />
            </svg>
            Free Updates for 2 year
          </li>
        </ul>
        <a
          className="chakra-link css-0"
          href="#"
        >
          <button
            className="chakra-button css-1sjc8a"
            type="button"
          >
            Buy now
          </button>
        </a>
      </div>
    </div>
  </section>
</div>
`;
